detection-rules/open_redirect_pirlsandiego.yml

```diff
@@ -0,0 +1,36 @@
+name: "Open Redirect: PIRL San Deigo"
+description: |
+  Message contains use of the PIRL San Deigo open redirect. This has been exploited in the wild.
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.root_domain == "pirlsandiego.net"
+          and strings.icontains(.href_url.path, '/LinkClick.aspx')
+          and strings.icontains(.href_url.url, 'link=')
+  )
```

Comment:
```suggestion
  )
  and sender.email.domain.root_domain != 'pirlsandiego.net'
```

Reply:
`pirlsandiego.net` doesn't have an MX record, so they must not be sending email from that.

Based on what I could find on their website, this is used to redirect users out of pirlsandiego.net, lots of examples on this page: http://pirlsandiego.net/Npirl/ContactInfo/tabid/541/Default.aspx

I'm confident it'll be abnormal to see this redirect in messages.

If we want to add confidence, we might be able to negate links where `tabid` and `mid` are present within the URL query params, which hasn't been observed in malicious use but is present on the website.  If we start seeing FPs, I'll get that added in quickly.

Reply:
Makes sense, thanks for the context 🫡